xoinvader/tests/test_animation.py

```diff
@@ -0,0 +1,105 @@
+import time
+
+import pytest
+
+from xoinvader.animation import Animation, AnimationManager
+
+
+class GameObject(object):
+    """Game object mock."""
+
+    anim_props = dict(
+        name="test-animation",
+        attr="attr", keyframes=[
+            (0.0, 1),
+            (0.5, 2),
+            (2.0, 10),
+        ])
+
+    dummy = dict(
+        name="dummy-animation",
+        attr="attr"
+    )
+
+    def __init__(self):
+        self.attr = 0
+
+
+def test_animation():
+
+    def keyframe_value(idx):
+        """Return value by idx."""
+        return GameObject.anim_props["keyframes"][idx][1]
+
+    with pytest.raises(ValueError):
+        Animation(bind=None, keyframes=[], **GameObject.dummy)
+
+    obj = GameObject()
+    anim = Animation(bind=obj, **obj.anim_props)
```

Comment:
This doesn't look like real-world scenario.
Why don't just create animation keyframe list in place and send it to Animation object?
It will make the test way more easy to understand.